interview-questions.md

```diff
@@ -9,49 +9,57 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
-2. What is React? Why would you use it?
 
-  Your answer:
+2. What is React? Why would you use it?
 
-  Researched answer:
+  Your answer: React is what developers use to create user interactive web pages. React uses components that hold data in a coding language called JSX. The components come together to build the webpages. 
 
+  Researched answer: React is a Javascript Library, created by Facebook in 2011, used to create user wed applications. React  React is component based. Components are reusable code that act like javascript functions and return HTML that is displayed on the user interface. 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The lifecycle method of a React class component is the constructor method and render method
 
-  Researched answer:
+  Researched answer: The lifecycle method that is required in a react class component is the render method that occurs in two of the three lifecycle stages of a class component, the mounting stage and the updating stage. the render method is the only method required as it holds the data that will be displayed in the user interface. This data is held in fragment tags. 
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a coding language that is a hybrid of javascript and HTML syntax. One syntax differnce in JSX from html is that tags have to be held on within fragment tags. 
 
-  Researched answer:
+  Researched answer: JSX is a JavaScript extension that allow you to write HTML and Javascript together. One difference in the syntax is that within JSX you have to hold elements within fragment tags, called a parent element. In HTML you can list as many elements as you would like without holding them in a fragment tag. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager that contains sharable code. The file that are modified in react when yarn is run is App.js. 
 
-  Researched answer:
+  Researched answer: Yarn is a package manager that allows you to share code with others. It allows you to troubleshoot more easily as you are able to use solution found by other developers. 
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: An anonymous function is a function without a name.
 
-  Researched answer:
+  Researched answer: An anonymous function is a function that does not have a name instead it take on the keyword function. anonymous functions do not have to be held in a seperate file which helps simplify the code.
```

Comment:
Such a great improvement in these answers, Will!  Your answer to #1 was exceptional!  I encourage you to use examples any chance you get.  It really shows your understanding.  When you come across something you don't know or are unsure of how to define, throw out examples => when you've used it, any pros or cons, etc.  Great work!